### PR TITLE
Improve documentation of key exchange mode configuration

### PR DESCRIPTION
**Pull request by hanno-becker**
This PR renames the TLS 1.3 specific key exchange configuration macros `KEY_EXCHANGE_MODE_XXX` to belong to the Mbed TLS namespace `MBEDTLS_YYY`. Moreover, it extends the documentation of the new API `mbedtls_ssl_conf_ke()` for the configuration of supported key exchange modes in (D)TLS 1.3.

### CODE REVIEW

**Comment by hanno-becker** [comment]
Oh, I just noted that the documentation enhancement already talks about the key exchange modes being defined as bitflags - that's not yet done and will be a subsequent PR.

File: include/mbedtls/ssl.h [hunk 1]
```diff
@@ -3223,19 +3223,66 @@ int mbedtls_ssl_conf_own_cert( mbedtls_ssl_config *conf,
...
+ *
+ * Users can specify which ciphersuites to support in (D)TLS versions 1.2
+ * and earlier via mbedtls_ssl_conf_ciphersuites(), and should refer to the
+ * corresponding documentation for more information.
+ *
+ * In (D)TLS 1.3, the former ciphersuite concept has been modified to allow
+ * separate negotiation of key establishment mode and symmetric encryption mode.
+ *
+ * In the lingo of (D)TLS 1.3, it is only the latter that is called
+ * 'ciphersuite', and it's configuration continues to be done through
```
Comment (gbryant-arm) [inline include/mbedtls/ssl.h]: its

Comment (hanno-becker) [reply include/mbedtls/ssl.h]: Fixing.

File: include/mbedtls/ssl.h [hunk 2]
```diff
@@ -3223,19 +3223,66 @@ int mbedtls_ssl_conf_own_cert( mbedtls_ssl_config *conf,
...
+ *         but excludes pure-ECDHE key exchanges.
+ *       - MBEDTLS_SSL_TLS13_KEY_EXCHANGE_MODE_ECDHE_ALL
+ *         Includes both pure-ECDHE and combined PSK-ECDHE key exchanges,
+ *
+ * \note If a PSK-based key exchange mode shall be supported, applications
+ *       must also use the APIs mbedtls_ssl_conf_psk() or
+ *       mbedtls_ssl_conf_psk_cb() or mbedtls_ssl_conf_psk_opaque()
+ *       to configure the PSKs to be used.
+ *
+ * \note If a ECDHE-based key exchange mode shall be supported, server-side
```
Comment (gbryant-arm) [inline include/mbedtls/ssl.h]: an

Comment (hanno-becker) [reply include/mbedtls/ssl.h]: Fixing.

**Comment by hanno-becker** [comment]
Thanks @gbryant-arm, I've fixed the typos you spotted. Can you please re-review?

**Comment by hannestschofenig** [comment]
Looks good to me.

**Pull request merged**